app/javascript/action_completed_button.js

```diff
@@ -0,0 +1,61 @@
+import CSRF from 'csrf'
+import { toast } from './vanillaToast'
+
+document.addEventListener('DOMContentLoaded', function () {
+  const buttons = document.querySelectorAll('.check-button')
+
+  buttons.forEach((button) => {
+    const commentableId = button.dataset.commentableId
+
+    if (!commentableId) return
```

Comment:
`button`が有効か否かでreturnした方が早いかもしれません。
また、`buttons`から`button`にすると、`forEach()`がいらなくなります〜。（配列ではなくなるため）
```suggestion
  if (!button) {
    return
  }
```

一応`grep`で確認したところ、`check-button`というクラス名はここでしか使われてないので、十分一意性が確保できるかなと思います。（このページでだけこのJSが機能するようになる。）
```
grep 'check-button' -r app/
app//components/action_completed_button_component.html.slim:          button.a-button.is-sm.is-block.check-button.is-muted-borderd(
app//components/action_completed_button_component.html.slim:          button.a-button.is-sm.is-block.check-button.is-warning(
app//javascript/action_completed_button.js:  const buttons = document.querySelectorAll('.check-button')
```

上のコメントと合わせて、以下のような提案をさせていただいています🙏
```js
document.addEventListener('DOMContentLoaded', function () {
  const button = document.querySelector('.check-button')
  if (!button) {
    return
  }
  const commentableId = button.dataset.commentableId
  button.addEventListener('click', function () {
// (略)
```

Reply:
https://github.com/fjordllc/bootcamp/pull/8488#discussion_r2045908184
こちらのコメントと同様、querySelectorとした場合動作しないように思えますがいかがでしょうか？

Reply:
一意性の確認までしていただきありがとうございます<( _ _ )>
こちらも修正いたしました